django_cloud_deploy/tool_requirements.py

```diff
@@ -0,0 +1,282 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Checks the user has the necesarry requirements to run the tool."""
+
+import os
+import shutil
+import stat
+import subprocess
+import sys
+import urllib.request
+
+from typing import List
+
+from django_cloud_deploy.cli import io
+
+
+class UnableToAutomaticallyInstallError(Exception):
+    """Thrown when the handle function is unable to install the requirement."""
+    pass
+
+
+class MissingRequirementError(Exception):
+    """Thrown when a requirement is missing.
+
+    Attributes:
+        name: Name of the requirement.
+        how_to_install_message: Information to help user to install
+            the requirement.
+
+    """
+
+    def __init__(self, name: str, how_to_install_message: str):
+        """Provides information on how to manually install the requirement.
+
+        Args:
+            name: Name of the requirement.
+            how_to_install_message: Information to help user to install
+                the requirement.
+        """
+        self.name = name
+        self.how_to_install_message = how_to_install_message
+
+
+class MissingRequirementsError(Exception):
+    """Thrown when one or more requirements are missing.
+
+    This is used by the caller to provide information to the user on how to
+    manually install missing requirements.
+
+    Attributes:
+        missing_requirements: List of missing requirements.
```

Comment:
Would you mention the type? I would have assumed string here.

Reply:
Looks like the type is mentioned in the constructor.